### PR TITLE
fix: php 8.1 notices

### PR DESCRIPTION
**Pull request by joostdebruijn**
This PR fixes some PHP 8.1 deprecations. I assume there are more occurences where the same deprecations are triggered, but those are the ones I encountered.

### CODE REVIEW

File: include/tcpdf_fonts.php [hunk 1]
```diff
@@ -2000,10 +2000,10 @@ public static function getUniord($uch) {
 	public static function UTF8StringToArray($str, $isunicode, &$currentfont) {
 		if ($isunicode) {
 			// requires PCRE unicode support turned on
-			$chars = TCPDF_STATIC::pregSplit('//','u', $str, -1, PREG_SPLIT_NO_EMPTY);
+			$chars = TCPDF_STATIC::pregSplit('//','u', $str ?? '', -1, PREG_SPLIT_NO_EMPTY);
```
Comment (williamdes) [inline include/tcpdf_fonts.php:2003]: You can not use this syntax, is is not old PHP 5 compatible :/

Comment (joostdebruijn) [reply include/tcpdf_fonts.php]: Ah! Not realizing that supporting PHP 5 was a requirement, fixed!

**Comment by laoneo** [comment]
Please merge this one and make a release to be able to run tcpdf on PHP 8.1.

**Review by laoneo — APPROVED** [review]

**Comment by williamdes** [comment]
> Please merge this one and make a release to be able to run tcpdf on PHP 8.1.

By the way tcpdf runs correctly (kind of) with PHP 8.1
If you add some new examples in the examples directory it will help catching the missing parts :)
And this is very important, add as much examples as possible to cover the existing code

**Review by williamdes — APPROVED** [review]

**Comment by laoneo** [comment]
It does, but throws some deprecate notices which are fixed with this pr.

**Review by famoser — APPROVED** [review]

**Review by nicolaasuni — APPROVED** [review]

**Pull request merged**